src/Plugin/EdgeKeyTypeBase.php

```diff
@@ -58,8 +58,11 @@ public function getAuthenticationType(KeyInterface $key): string {
   /**
    * {@inheritdoc}
    */
-  public function getEndpoint(KeyInterface $key): string {
-    return $key->getKeyValues()['endpoint'] ?? Client::DEFAULT_ENDPOINT;
+  public function getEndpoint(KeyInterface $key, bool $safeReturn = FALSE): ?string {
+    if ($safeReturn) {
```

Comment:
```
if (array_key_exist(`endpoint`, $key->getKeyValues()) {
  return $key->getKeyValues()['endpoint'];
}

return $safeReturn ? Client::DEFAULT_ENDPOINT : NULL;
```

Just because I can see two identical `$key->getKeyValues()['endpoint'] ??` checks in your implementation.